app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,70 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class TaskServiceTest {
+
+    private TaskService taskService;
+
+    private final static String NEW_TITLE = "new task";
+    private final static String TITLE_POSTFIX = " spring";
+
+    @BeforeEach
+    void setUp() {
+        taskService = new TaskService();
+
+        Task source = new Task();
+        source.setTitle(NEW_TITLE);
+        Task task = taskService.createTask(source);
+    }
+
+    @Test
+    void getTasks() {
+        List<Task> tasks = taskService.getTasks();
+        assertThat(tasks).hasSize(1);
+    }
+
+    @Test
+    void getTask() {
+        assertThatThrownBy(() -> taskService.getTask(100L))
+                .isInstanceOf(TaskNotFoundException.class);
+    }
+
+    @Test
+    void createTask() {
+        Task source = new Task();
+        source.setTitle(NEW_TITLE + TITLE_POSTFIX);
+        Task task = taskService.createTask(source);
+
+        assertThat(task.getId()).isEqualTo(2L);
+        assertThat(task.getTitle()).isEqualTo(NEW_TITLE + TITLE_POSTFIX);
+    }
+
+    @Test
+    void updateTask() {
+        Task source = new Task();
+        source.setTitle(NEW_TITLE + TITLE_POSTFIX);
+        Task task = taskService.updateTask(1L, source);
+
+        assertThat(task.getId()).isEqualTo(1L);
+        assertThat(task.getTitle()).isEqualTo(NEW_TITLE + TITLE_POSTFIX);
+
+    }
+
+    @Test
+    void deleteTask() {
+        Task task = taskService.deleteTask(1L);
+        List<Task> tasks = taskService.getTasks();
+
+        assertThat(task.getId()).isEqualTo(1L);
+        assertThat(tasks).hasSize(0);
```

Comment:
둘이 테스트하는 대상이 조금 다른 것 같아서, 분리해주면 좋을 것 같아요. 
```suggestion
        Task task = taskService.deleteTask(1L);
        
        assertThat(task.getId()).isEqualTo(1L);
        
        List<Task> tasks = taskService.getTasks();

        assertThat(tasks).hasSize(0);
```

혹은 위의 테스트처럼 무엇이 변화했는지로 의도를 드러내도 좋을 것 같아요. 테스트를 작성했을 때 이 코드가 무엇이냐 보다는 이 코드를 왜 사용해야 하느냐가 드러나도록 하면 좋아요. 이 코드를 사용하면 어떤 일이 벌어져 라고 말하는 것을 생각해보시면 좋을 것 같아요. 이러한 관점은 의도를 드러내는데 매우매우매우매우매우 중요하니 많이 고민해 보시면 좋을 것 같아요. 왜냐하면 테스트 코드 뿐만 아니라 코드에서도 마찬가지거든요

```java
        int oldSize = taskService.getTasks();

        Task task = taskService.deleteTask(1L);
        
        assertThat(task.getId()).isEqualTo(1L);
        
        List<Task> tasks = taskService.getTasks();
        int newSize = tasks.size;

        assertThat(oldSize - newSize).hasSize(1);
```

Reply:
검증하는 부분을 묶었는데 어떤 코드로 무엇이 변화해서 검증하는지 바로 알기가 힘들 수 도 있겠네요.

할일을 생성하는 코드를 사용하는 이유 중 하나가 할일 목록에 추가하는 것이기 때문에

아래 코드가 맘에 들었어요.